src/diagnostics/MOM_wave_speed.F90

```diff
@@ -830,59 +930,69 @@ subroutine wave_speeds(h, tv, G, GV, US, nmodes, cn, CS, w_struct, u_struct, u_s
   w_struct(:,:,:,:) = 0.0
 
   min_h_frac = tol_Hfrac / real(nz)
-  !$OMP parallel do default(private) shared(is,ie,js,je,nz,h,G,GV,US,CS,min_h_frac,use_EOS, &
-  !$OMP                                     Z_to_pres,tv,cn,g_Rho0,nmodes,cg1_min2,better_est, &
-  !$OMP                                     tol_solve,tol_merge,c2_scale)
+  !$OMP parallel do default(private) shared(is,ie,js,je,nz,h,G,GV,US,CS,use_EOS,nonBous, &
+  !$OMP                                     min_h_frac,H_to_pres,tv,cn,g_Rho0,nmodes,cg1_min2, &
+  !$OMP                                     better_est,tol_solve,tol_merge,c2_scale)
   do j=js,je
     !   First merge very thin layers with the one above (or below if they are
     ! at the top).  This also transposes the row order so that columns can
     ! be worked upon one at a time.
     do i=is,ie ; htot(i) = 0.0 ; enddo
-    do k=1,nz ; do i=is,ie ; htot(i) = htot(i) + h(i,j,k)*GV%H_to_Z ; enddo ; enddo
+    do k=1,nz ; do i=is,ie ; htot(i) = htot(i) + h(i,j,k) ; enddo ; enddo
+
+    call thickness_to_dz(h, tv, dz_2d, j, G, GV)
```

Comment:
`SpV_avg` has zero values where `h` is non-zero, results in `dzc=0` at denominator further down and crashes. it may need an update before call to `thickness_to_dz`

Reply:
The problem is actually more extensive than this observation suggests, because the MOM_internal_tides code is expecting certain calculations to occur at all points in the entire data domain, but the thickness_to_dz call here is only working on the computational domain.  I think that the real answer here is to replace the full_halos argument to wave_speeds with a halo_size argument like everywhere else in the MOM6 code, and then only do the internal_tides calculations on a finite halo size, rather than basing its range of calculations on the size of the data domain.

Reply:
+1 on cleaning the halos in `internal_tides` but I also think the specific volumes have problematic zeros in the computational domain. I modified `thickness_to_dz_jslice` of `MOM_interface_heights.F90` with an exception if `SpV_avg` is zero while `h` is not:

```fortran
  if ((.not.GV%Boussinesq) .and. allocated(tv%SpV_avg))  then
    if ((allocated(tv%SpV_avg)) .and. (tv%valid_SpV_halo < halo)) then
      if (tv%valid_SpV_halo < 0) then
        mesg = "invalid values of SpV_avg."
      else
        write(mesg, '("insufficiently large SpV_avg halos of width ", i2, " but ", i2," is needed.")') &
                     tv%valid_SpV_halo, halo
      endif
      call MOM_error(FATAL, "thickness_to_dz called in fully non-Boussinesq mode with "//trim(mesg))
    endif

    do k=1,nz ; do i=is,ie
      if ((h(i,j,k) > 0.) .and. (tv%SpV_avg(i,j,k) == 0.)) then
        call MOM_error(FATAL, "Specific volume is zero on non-zero layer")
      else
        dz(i,k) = GV%H_to_RZ * h(i,j,k) * tv%SpV_avg(i,j,k)
      endif
    enddo ; enddo
  else
    do k=1,nz ; do i=is,ie
      dz(i,k) = GV%H_to_Z * h(i,j,k)
    enddo ; enddo
  endif
```

and I'm catching `Specific volume is zero on non-zero layer` errors. I'm not passing a halo argument so the computation should only happen on the computational domain.